xds/bootstrap/bootstrap.go

```diff
@@ -58,6 +59,9 @@ func RegisterCredentials(c Credentials) {
 // GetCredentials returns the credentials associated with a given name.
 // If no credentials are registered with the name, nil will be returned.
 func GetCredentials(name string) Credentials {
+	if name == "jwt_token_file" && !envconfig.XDSBootstrapCallCredsEnabled {
```

Comment:
FYI, since the convention is to register the credentials via an `init()` function in `xds/bootstrap/credentials.go`, it's not possible to opt in or out of the registration there and at the same time to be able to unit test if the feature is guarded by `XDSBootstrapCallCredsEnabled`. Hence why the check has been moved here.

Reply:
The environment variable should only guard whether we process the new `call_creds` field in the bootstrap configuration. It should not guard the registration of the JWT call creds implementation.